site/gdocs/utils.ts

```diff
@@ -152,12 +153,22 @@ export const useLinkedChart = (
         }
     }
 
+    const resolvedUrl = (() => {
+        const parsedResolvedUrl = Url.fromURL(linkedChart.resolvedUrl)
+        if (!queryString) return parsedResolvedUrl.fullUrl
+
+        return parsedResolvedUrl.setQueryParams({
+            ...parsedResolvedUrl.queryParams,
+            ...parsedOriginalUrl.queryParams,
+        }).fullUrl
+    })()
```

Comment:
I get the slight benefit of using an IIFE here (namely, that you don't need to have mutable vars), but it's not a pattern we normally use.

I think it's fine to pull this out into "normal" variables, or alternatively into a scoped `resolveRedirectQueryParams`-or-so function.